xdsl/ir.py

```diff
@@ -54,8 +57,35 @@ class MLContext:
     registered_unregistered_ops: dict[str, type[Operation]] = field(
         default_factory=dict)
 
+    def _register_module(self, dialect: ModuleType):
```

Comment:
I think this should be renamed, otherwise it's not clear whether you are talking about an IR Module OP (that might contain some IRDL code) or a python module.